tests/functional/test_application_restricted.py

```diff
@@ -180,94 +179,100 @@ def auth():
 
 @given("I create a new app")
 def create_test_app(setup_session, context):
-    product, app, token = setup_session
+    product, app, token_response, developer_apps, api_products = setup_session
     context["product"] = product
     context["app"] = app
-    context["token"] = token
+    context["token_response"] = token_response
+    context["developer_apps"] = developer_apps
+    context["api_products"] = api_products
 
 
 @given(parsers.parse("I add the attribute with key of {key} and a value of {value}"))
 def add_custom_attribute_to_app(key: str, value: str, context: dict):
 
     app = context["app"]
-
-    asyncio.run(
-        app.set_custom_attributes(
-            {"jwks-resource-url": config.JWKS_RESOURCE_URL, key: value}
-        )
+    developer_apps = context["developer_apps"]
+    app.set_custom_attributes(
+        {"jwks-resource-url": config.JWKS_RESOURCE_URL, key: value},
+        developer_apps
```

Comment:
I know this isn't your doing. But this is an inappropriately named feature/function. This is overwriting not adding to the custom attributes, no? Which, if an ASID is in the attributes, will be lost, right?

Reply:
Might be worth adding to the refactoring ticket?

Reply:
I'll refactor to, at least, keep an ASID in there. I'll mention it in SPINEDEM-1680 as well.